File: code/test1a1.R
##########################################################################
# User: Hirotaka Miura
# Position: Doctoral Student
# Organization: MIT Sloan
##########################################################################
# 08/10/2022: Modified.
# 08/09/2022: Previously modified.
# 08/09/2022: Created.
# Description: 
#   - Test program.
# Modifications:
#   08/10/2022:
#     - Remove parallel components.
##########################################################################

#########
# Setup #
#########

# Load and execute preamble function.
source("preamble.R")
preamble()
# Read command line argument(s).
args=commandArgs(trailingOnly=TRUE)
# Set program name.
argsi=1; program=args[argsi]
argsi=argsi+1; progver=args[argsi]
# Load and execute function to display job info.
source("jobinfo.R")
jobinfo(program=program,progver=progver)
# Load user defined module for utility functions.
source("util.R")
# Get input parameters.
argsi=argsi+1; params=eval(parse(text=args[argsi]))
# Show input parameters.
cat("\n***** Input parameters:\n")
print(params)

#-------------------------------
# DO SOMETHING HERE
#===============================

# Display section title.
cat("\n\n*****")
cat("\n***** DO SOMETHING HERE")
cat("\n*****\n\n")

###########
# Wrap-up #
###########
# Display job information.
jobinfo(program=program,progver=progver,disp_pack=1)

File: code/test1a1.py
##########################################################################
# User: Hirotaka Miura
# Position: Doctoral Student
# Organization: MIT Sloan
##########################################################################
# 08/10/2022: Modified.
# 08/10/2022: Previously modified.
# 08/10/2022: Created.
# Description: 
#   - Test program.
# Modifications:
##########################################################################

# Load modules.
import sys                      # For system functions.
import util                     # Custom utility functions.

# Get current program name.
progname=sys.argv[0].split('.')[0]
# Show job information.
util.show_job_info(progname=progname)
# Set preamble.
util.preamble()

#---- Evaluate input parameters ----#
print('***** Evaluate input parameters')
# Evaluate.
params=eval(sys.argv[1])
# Display input parameters.
for pkey,pval in params.items():
  print(pkey,": ",pval,sep="")

#-------------------------------
# DO SOMETHING HERE
#===============================

# Display message.
print('\n*****')
print('***** DO SOMETHING HERE')
print('*****')

#-------------------------------
# WRAP-UP
#===============================

# Show job information.
util.show_job_info(progname=progname,disp_packages=1)
